src/custom/hooks/useFetchProfile.ts

```diff
@@ -3,22 +3,40 @@ import { useEffect, useState } from 'react'
 import { getProfileData } from 'api/gnosisProtocol'
 import { ProfileData } from 'api/gnosisProtocol/api'
 
-export default function useFetchProfile() {
+type FetchProfileState = {
+  profileData: ProfileData | null
+  //error: string
+  isLoading: boolean
+}
+
+const emptyState: FetchProfileState = {
+  profileData: null,
+  //error: '',
+  isLoading: false,
+}
+
+const FETCH_INTERVAL_IN_MINUTES = 5
+
+export default function useFetchProfile(): FetchProfileState {
   const { account, chainId } = useActiveWeb3React()
-  const [profileData, setProfileData] = useState<ProfileData | null>(null)
+  const [profile, setProfile] = useState<FetchProfileState>(emptyState)
 
   useEffect(() => {
     async function fetchAndSetProfileData() {
       if (chainId && account) {
+        setProfile({ ...emptyState, isLoading: true })
         const profileData = await getProfileData(chainId, account)
-        setProfileData(profileData)
+        setProfile({ ...emptyState, isLoading: false, profileData })
```

Comment:
Any reason to reset the state on every load?

I would prefer to have the old value visible while it's loading

Reply:
Yes, good point, it even added an annoying flash between the dash and the value while loading